Sentry integration (#202)
avantifellows/plio-backend: docs/ERROR-MONITORING.md

## Error Monitoring

If you're setting up Plio for staging or production, you may also wish to capture errors faced by users in real-time. For that purpose, Plio uses [Sentry.io](https://sentry.io/) for real-time error monitoring and reporting.

### Pre-requisites
1. Create an account at Sentry.
2. Set up your organization.
3. Create a new Django project. Name it `plio-backend`.

### Enable error logging
1. Set up env variable for `SENTRY_DSN`. You can get the value of DSN for your projects from `Project Settings > SDK Setup > Client Keys (DSN)`

> 2. Make sure your `NODE_ENV` is either 'staging' or 'production'.

Author: Sorry, updated!

> 3. Trigger an error by adding the following line in your `urls.py` file:

Reviewer: ```suggestion
3. Test your set up by triggering an error - add the following line in your `urls.py` file:
```

    ```py
    from django.urls import path
    def trigger_error(request):
        division_by_zero = 1 / 0

    urlpatterns = [
        path('trigger-error/', trigger_error),
        # other paths
    ]
    ```

> 4. You should now see the error at your Sentry dashboard as well from `Issues` tab.

Reviewer: one should only get an error if the trigger-error endpoint is called right?

Author: Yes. But you may receive errors on existing endpoints also if there are any. The code covers all endpoints.

> 4. You should now see the error at your Sentry dashboard as well from `Issues` tab.

Reviewer: ```suggestion
4. You should now see the error at your Sentry dashboard as well from the `Issues` tab.
```

### Enable reporting
By default, Sentry sends error reports to your inbox. However, you may prefer other channels to receive error notifications.

At Plio, we use Discord to receive real-time alerts using Sentry Webhooks. If you're also using Discord, or a different tool that supports webhooks, follow the steps below to start receiving alerts:

> 1. Create a new channel at Discord where you want to receive the alerts. Then, enable the webhook and copy webhook url. `Channel settings > Integrations > Webhook`

Reviewer: ```suggestion
1. Create a new channel at Discord where you want to receive the alerts. Then, enable the webhook and copy the webhook url. `Channel settings > Integrations > Webhook`
```

> 2. Create a SentryDiscord webhook.

Reviewer: this has been done for staging right? I still need to set another one for prod right? And I should be able to test it locally by setting APP_ENV and SENTRY_DSN appropriately?

Author: This hasn't been done for staging or production as the code deployment was pending.

Yes, you can test it by setting APP_ENV and SENTRY_DSN.
